### PR TITLE
[notifications][docs] fix broken links and type issues

### PR DESCRIPTION
**Pull request by Simek**
# Why

There were few broken links and missing types issues in the notifications docs.

Screenshot 2024-10-28 at 20 50 40

# How

Summary of the changes:
* correct broken links and type names in package doc comments
* export `MaybeNotificationResponse`
* correct broken links in MDX files for unversioned and SDK 52
* fix duplicated header on unversioned and SDK 52 pages
* skip linking of global or manually hidden types

> [!note]
> The only one issue left is with `SetBadgeCountOptions`, which I was not confident to refactor/fix without more testing. Looks like we have two types names like that, one is local, other is exported, but not listed in main file exports.

# Test Plan

On the package side tests and lint checks are passing.

On the docs front, changes have been reviewed by running app locally. All tests and lint checks are passing.

### CODE REVIEW

File: docs/pages/versions/unversioned/sdk/notifications.mdx [hunk 1]
```diff
@@ -432,23 +432,23 @@ Learn how to configure the native projects in the [installation instructions in
```
Comment (douglowder) [inline docs/pages/versions/unversioned/sdk/notifications.mdx:432]: <i class='graphite__hidden'>[Re: line +22]<br/><br/></i>

Hi @vonovak I believe we are just doing warnings in SDK 52 and disabling this feature in Expo Go for SDK 53, correct?<span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/expo/expo/32427?utm_source=unchanged-line-comment">Graphite</a>.</span>

Comment (vonovak) [reply docs/pages/versions/unversioned/sdk/notifications.mdx:432]: fixed it

**Review by vonovak — APPROVED** [review]
Thank you!

**Comment by vonovak** [comment]
I will apply these to the 52 docs too and then merge

**Pull request merged**